crates/ere-sp1/src/compile.rs

```diff
@@ -9,37 +10,50 @@ use tracing::info;
 
 use crate::error::CompileError;
 
-pub fn compile(guest_program_full_path: &Path) -> Result<Vec<u8>, CompileError> {
+pub fn compile(
+    workspace_directory: &Path,
+    guest_program_relative: &Path,
+) -> Result<Vec<u8>, CompileError> {
     // Build the SP1 docker image
     let tag = "ere-build-sp1:latest";
     docker::build_image(&PathBuf::from("docker/sp1/Dockerfile"), tag)
         .map_err(|e| CompileError::DockerImageBuildFailed(Box::new(e)))?;
 
     // Compile the guest program using the SP1 docker image
-    let guest_program_path_str = guest_program_full_path
+    let mount_directory_str = workspace_directory
         .to_str()
-        .ok_or_else(|| CompileError::InvalidGuestPath(guest_program_full_path.to_path_buf()))?;
+        .ok_or_else(|| CompileError::InvalidMountPath(workspace_directory.to_path_buf()))?;
+
     let elf_output_dir = TempDir::new().map_err(CompileError::CreatingTempOutputDirectoryFailed)?;
     let elf_output_dir_str = elf_output_dir
         .path()
         .to_str()
         .ok_or_else(|| CompileError::InvalidTempOutputPath(elf_output_dir.path().to_path_buf()))?;
 
-    info!("Compiling program: {}", guest_program_path_str);
+    let container_mount_directory = PathBuf::from_str("/guest-workspace").unwrap();
+    let container_guest_program_path = container_mount_directory.join(guest_program_relative);
+    let container_guest_program_str = container_guest_program_path
+        .to_str()
+        .ok_or_else(|| CompileError::InvalidGuestPath(guest_program_relative.to_path_buf()))?;
+
+    info!(
+        "Compiling program: mount_directory={} guest_program={}",
+        mount_directory_str, container_guest_program_str
+    );
 
     let status = Command::new("docker")
```

Comment:
The two main changes for SP1:
- The `-v` mounting is the workspace directory.
- The argument to the compiler is the guest relative path (since from the compiler perspective the workdir is the mounted folder).